gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/JobExecutionFuture.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.runtime.api;
+
+import java.util.concurrent.Future;
+
+/**
+ * A simple monitoring and future object. This object can be used as a normal {@link Future} object to get
+ * {@link ExecutionResult}. Meanwhile it can also get an object ({@link JobExecutionMonitor}) to monitor current job running status.
+ */
+public interface JobExecutionFuture extends Future<ExecutionResult> {
```

Comment:
Can we move the future into the monitor and have the monitor have the ability to monitor and get the final result?